aws/container-linux/kubernetes/network.tf

```diff
@@ -52,6 +52,60 @@ resource "aws_subnet" "public" {
 resource "aws_route_table_association" "public" {
   count = "${length(data.aws_availability_zones.all.names)}"
 
-  route_table_id = "${aws_route_table.default.id}"
+  route_table_id = "${aws_route_table.public.id}"
   subnet_id      = "${element(aws_subnet.public.*.id, count.index)}"
 }
+
+resource "aws_subnet" "private" {
+  count = "${length(data.aws_availability_zones.all.names)}"
+
+  vpc_id            = "${aws_vpc.network.id}"
+  availability_zone = "${data.aws_availability_zones.all.names[count.index]}"
+
+  cidr_block                      = "${cidrsubnet(var.host_cidr, 4, count.index + 8)}"
+  ipv6_cidr_block                 = "${cidrsubnet(aws_vpc.network.ipv6_cidr_block, 8, count.index + 8)}"
+  assign_ipv6_address_on_creation = true
+
+  tags = "${map("Name", "${var.cluster_name}-private-${count.index}")}"
+}
+
+resource "aws_route_table" "private" {
+  vpc_id = "${aws_vpc.network.id}"
+
+  route {
+    cidr_block     = "0.0.0.0/0"
+    nat_gateway_id = "${aws_nat_gateway.nat.id}"
+  }
+
+  route {
+    ipv6_cidr_block        =   "::/0"
+    egress_only_gateway_id = "${aws_egress_only_internet_gateway.egress_igw.id}"
+  }
+
+  tags = "${map("Name", "${var.cluster_name}-private")}"
+}
+
+resource "aws_route_table_association" "private" {
+  count = "${length(data.aws_availability_zones.all.names)}"
+
+  route_table_id = "${aws_route_table.private.id}"
+  subnet_id      = "${element(aws_subnet.private.*.id, count.index)}"
+}
+
+
+resource "aws_eip" "nat" {
```

Comment:
nat?

Reply:
I think you may need the EIP to explicitly depend on the IGW

Reply:
> nat?

Can you expand?

> I think you may need the EIP to explicitly depend on the IGW

Backwards (gateways use an EIP and need it allocated first). The egress gateway is actually for ipv6 and the NAT gateway is for ipv4.

Reply:
>Can you expand?

why name the elastic ip nat?

Reply:
>Backwards (gateways use an EIP and need it allocated first). The egress gateway is actually for ipv6 and the NAT gateway is for ipv4.

from my understanding an epi address is accessible via the igw and not the other way around

Reply:
Oh. Because it's going to be associated with the NAT gateway. I don't need to reference it anywhere else.

Reply:
Sorry, missing a word up there. The _NAT_ gateway consumes the EIP. The _Internet_ gateway allows instances in the public subnet (including the NAT) to access the internet. 

I believe the confusing bit is that the internet gateway provides NAT for instances in the public subnet. And the NAT gateway and egress-only internet gateway provide routing for instances within the private subnet.